.github/workflows/build_and_upload_web_files_staging.yml

```diff
@@ -0,0 +1,80 @@
+name: Build and upload web files - Staging
+
+on:
+  workflow_dispatch:
+  workflow_run:
+    workflows: ['Terraform apply - Staging']
+    types:
+      - completed
+  push:
+    branches:
+      - main
+
+env:
+  AWS_REGION: ca-central-1
+  TERRAFORM_VERSION: 1.11.2
+  TERRAGRUNT_VERSION: 0.78.0
+  NX_CLOUD_DISTRIBUTED_EXECUTION: true
+  NX_CLOUD_DISTRIBUTED_EXECUTION_AGENT_COUNT: 3
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Nx Cloud - Build web - Main Job
+    if: github.repository == 'cds-snc/cra-arc-upd-tbpc' && (github.event_name == 'push' || github.event.workflow_run.conclusion == 'success')
```

Comment:
I don't think you will need the repo name here, was there a specific reason to include it?